### PR TITLE
resolving issue #11376 

### LINKED ISSUE #11376: Fix: Follow button on blog page only works for the first post
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new** and looking through our general [help wanted](https://github.com/publiclab/plots2/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Claim the issue by commenting. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!
<details>
  <summary>💡 <i>Learn how to claim</i> 🙋</summary>
  <h3>Claiming an issue</h3>
  <p>Unless the issue is marked as reserved for someone, you can just say "I'd like to try this!" and then you've claimed it - no need to wait for someone to assign it to you. Just be sure you link your pull request (PR) to this issue so we can see where your solution is.</p>
  <p>And open one early if possible - even before you've completed it with additional commits - and others can help you figure out any issues you may face.</p>
</details>

- [ ] 📝 **Update** the file [app/views/tag/blog.html.erb](https://github.com/publiclab/plots2/blob/main/app/views/tag/blog.html.erb#L67) in the `plots2` repository (press the little pen Icon) and edit the line as shown below.

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
@@ -64,7 +64,7 @@
             <% end %>
           </small></p>
           <p><%= raw auto_link(insert_extras(node.latest.render_body), :sanitize => false) %></p>
-          <p><a class="btn btn-outline-secondary btn-md" href="<%= node.path %>" style="color: black"><%=raw translation('tag.blog.read_more')%> <i class="fa fa-angle-double-right"></i></a><%= render partial: 'tag/subscribe_button', locals:{tags: node.tags} %></p>
+          <p><a class="btn btn-outline-secondary btn-md" href="<%= node.path %>" style="color: black"><%=raw translation('tag.blog.read_more')%> <i class="fa fa-angle-double-right"></i></a><%= render partial: 'tag/subscribe_button', locals:{tags: node.tags, index: i} %></p>
           <p>
             <%= render :partial => "like/like", :locals => {:node => node, :tagnames => node.tags.collect(&:name) } %>
             <% node.tags[0..3].each do |tag| %>
```

- [ ] 📝 **Update** the file [app/views/tag/_subscribe_button.html.erb](https://github.com/publiclab/plots2/blob/main/app/views/tag/_subscribe_button.html.erb) in the `plots2` repository (press the little pen Icon) and edit the line as shown below.

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
@@ -1,21 +1,21 @@
 <!-- This button is used to display tags in a popover and subscribe to multiple tags using a single button -->
 <% buttonStyle ||= 'btn-outline-secondary btn-md' %>
 <% buttonColor ||= 'black' %>
-<a id="popover_button" class="btn <%= buttonStyle %>" style="color:<%= buttonColor %>;">Follow
+<a id="popover_button-<%= index %>" class="btn <%= buttonStyle %>" style="color:<%= buttonColor %>;">Follow
   <i class="fa fa-plus" style="color:<%= buttonColor %>;"></i>
 </a>

-<div id="mypopcontent" class="d-none" onmouseleave="show(this)">
+<div id="mypopcontent-<%= index %>" class="d-none" onmouseleave="show(this)">
   <%= render partial: "tag/tags_popover", locals: {tags: tags} %>
 </div>

 <script type="text/javascript">
-  $('#popover_button').popover({
-    content: $('#mypopcontent').clone().removeClass('d-none'),
+  $('#popover_button-<%= index %>').popover({
+    content: $('#mypopcontent-<%= index %>').clone().removeClass('d-none'),
     placement: 'bottom',
     html: true,
   });
   function show(elem){
-  $('#popover_button').popover("hide")
+  $('#popover_button-<%= index %>').popover("hide")
   }
 </script>
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://app.egghead.io/playlists/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there.

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming the `first-timers-only` issue. We're happy to keep it assigned to you as long as you need it if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance!

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it.

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below, and [reach out in our chatroom](https://publiclab.org/chat)!

### PR DESCRIPTION
**Pull request by vineethraik**
Fixes #11376  

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by vineethraik** [comment]
@publiclab/reviewers please review this PR

File: app/views/tag/_subscribe_button.html.erb [hunk 1]
```diff
@@ -1,21 +1,21 @@
 <!-- This button is used to display tags in a popover and subscribe to multiple tags using a single button -->
 <% buttonStyle ||= 'btn-outline-secondary btn-md' %>
 <% buttonColor ||= 'black' %>
```
Comment (TildaDares) [inline app/views/tag/_subscribe_button.html.erb:3]:
```suggestion
<% buttonColor ||= 'black' %>
<% index ||= 0 %>
```

File: app/views/tag/_subscribe_button.html.erb [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
-  $('#popover_button').popover({
-    content: $('#mypopcontent').clone().removeClass('d-none'),
+  $('#popover_button-<%= index %>').popover({
+    content: $('#mypopcontent-<%= index %>').clone().removeClass('d-none'),
     placement: 'bottom',
     html: true,
   });
   function show(elem){
-  $('#popover_button').popover("hide")
+    $('#popover_button-<%= index %>').popover("hide")
```
Comment (TildaDares) [inline app/views/tag/_subscribe_button.html.erb:20]:
```suggestion
  $('#popover_button-<%= index %>').popover("hide")
```

Comment (TildaDares) [inline app/views/tag/_subscribe_button.html.erb:20]: Thank you for indenting this @vineethraik!

**Review by TildaDares — APPROVED** [review]
Amazing job @vineethraik!!

**Comment by TildaDares** [comment]
Congratulations on merging your PR @vineethraik 🎉🎉

File: app/views/tag/_subscribe_button.html.erb (hunk 2 above)
Comment (vineethraik) [reply app/views/tag/_subscribe_button.html.erb:20]: Thank you for providing a chance to be a Open Source contributor. 
If you dont mind could you answer, Why did the check's failed  for the first run ?

**Comment by vineethraik** [comment]
> Congratulations on merging your PR @vineethraik 🎉🎉

Thank you for providing a chance to Contribute to Open Source.

File: app/views/tag/_subscribe_button.html.erb (hunk 2 above)
Comment (TildaDares) [reply app/views/tag/_subscribe_button.html.erb:20]: Hi @vineethraik, other files were using this partial but the `index` variable was not passed through those files as we did in `blog.html.erb`. So I added a check at the top of this page that assigns `index` to `0` if it's `undefined`. Please let me know if you need more explanations.

Sorry for not clarifying earlier!

Comment (vineethraik) [reply app/views/tag/_subscribe_button.html.erb:20]: Thank you  for the clarification.

**Pull request merged**